SDK/AppCenterPush/Microsoft.AppCenter.Push.UWP/Push.cs

```diff
@@ -74,19 +94,21 @@ private void ApplyEnabledState(bool enabled)
                     {
                         using (_mutex.GetLock(state))
                         {
-                            var pushToken = channel?.Uri;
-                            if (!string.IsNullOrEmpty(pushToken))
+                            LatestPushToken = channel?.Uri;
+                            if (!string.IsNullOrEmpty(LatestPushToken))
                             {
                                 // Save channel member
                                 _channel = channel;
 
-                                // Subscribe to push
+                                // Subscribe to UserId Change.
+                                UserIdContext.UserIdUpdated += OnUserIdUpdated;
```

Comment:
I think this should be removed and then added to avoid duplicate invocations.
```suggestion
                                UserIdContext.UserIdUpdated -= OnUserIdUpdated;
                                UserIdContext.UserIdUpdated += OnUserIdUpdated;
```